cmd/dockerd/options_test.go

```diff
@@ -17,9 +17,9 @@ func TestCommonOptionsInstallFlags(t *testing.T) {
 	opts.InstallFlags(flags)
 
 	err := flags.Parse([]string{
-		"--tlscacert=\"/foo/cafile\"",
-		"--tlscert=\"/foo/cert\"",
-		"--tlskey=\"/foo/key\"",
+		"--tlscacert=/foo/cafile",
+		"--tlscert=/foo/cert",
+		"--tlskey=/foo/key",
```

Comment:
Can the other format still be parsed?  I wouldn't expect to see a unit test change if there's no change in behavior.

Reply:
Sorry, I guess I should back up and explain this in more detail:

In https://github.com/moby/moby/issues/29761, it was discovered that after Docker changed from one flag parsing library to another, it lost a minor "feature" where it would strip quotes (outside whatever quote parsing the shell provides).  In shell syntax, we're talking about `docker '--flag="value"' ...` (note the extra double quotes inside that get passed verbatim).  The old flag library would quietly strip those, but the new flag library did not.

It turned out that `docker-machine` was _relying_ on that double stripping for how it was passing the three `--tls*` options to the Docker CLI, so this file was created as a wrapper to strip the quotes, but only from those three CLI flags.  In that original implementation, the CLI and Daemon were both sharing the same parsing code for those three TLS-related flags, so the daemon got support for stripping these quotes even though it wasn't actually needed on the daemon (`docker-machine` only passed them to the client).

Eventually, the CLI moved to a separate repository, and this code was copied over there, but stayed here as well even though it's not necessary in the daemon for the original bug, so I'm arguing that we should remove it.

The reason there's a unit test change here is because this test appears to be intended to make sure setting options works, but was also double-purposed to test that they would strip these _extra_ double quotes.  So for this change to move forward, either this test needs to strip the extra double quotes or validate that the value that comes in includes them (which if this PR goes forward would instead mean a relative path that literally starts with a double quote, so I don't think is terribly useful and I'd close this instead if that's the consensus).